handler/schedule.go

```diff
@@ -206,6 +233,80 @@ func (s *ScheduleHandler) getRandomSchedule(jobType k8upv1alpha1.JobType, origin
 }
 
 func (s *ScheduleHandler) setEffectiveSchedule(jobType k8upv1alpha1.JobType, schedule k8upv1alpha1.ScheduleDefinition) {
-	s.schedule.Status.EffectiveSchedules[jobType] = schedule
+	if s.effectiveSchedule == nil {
+		s.createNewEffectiveScheduleObj()
+	}
+	schedules := s.effectiveSchedule.Spec.EffectiveSchedules
+	schedules = append(schedules, k8upv1alpha1.JobRef{
+		Name:              s.schedule.Name,
+		Namespace:         s.schedule.Namespace,
+		JobType:           jobType,
+		GeneratedSchedule: schedule,
+	})
+	s.effectiveSchedule.Spec.EffectiveSchedules = schedules
+	s.requireStatusUpdate = true
+}
+
+func (s *ScheduleHandler) createNewEffectiveScheduleObj() {
+	newSchedule := &k8upv1alpha1.EffectiveSchedule{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cfg.Config.OperatorNamespace,
+			Name:      rand.String(32),
+		},
+	}
 	s.requireStatusUpdate = true
+	s.Log.Info("Creating new EffectiveSchedule", "name", k8upv1alpha1.GetNamespacedName(newSchedule))
+	err := s.Client.Create(s.CTX, newSchedule)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		s.Log.Error(err, "could not persist effective schedules", "name", newSchedule.Name)
+		s.SetConditionFalseWithMessage(k8upv1alpha1.ConditionReady, k8upv1alpha1.ReasonCreationFailed, "could not persist effective schedules: %s", err.Error())
+	} else {
+		s.SetConditionTrueWithMessage(k8upv1alpha1.ConditionReady, k8upv1alpha1.ReasonReady, "effective schedule created: '%s'", k8upv1alpha1.GetNamespacedName(newSchedule))
+		s.effectiveSchedule = newSchedule
+	}
```

Comment:
Removing the else clause means that the following code is executed regardless of the outcome in the if-condition, but that's not the intention here. I can only set true condition on successes. I would need a return statement in the if-clause to prevent that.

Reply:
You're right, that should have actually be part of my suggestion.

I suggest to return at the end of the `if` case and not use the `else` case here. it reduces one branch in the code – if not actually, then at least mentally.

Reply:
after last dicussions I might need to rework this anyway